.gitignore

```diff
@@ -17,6 +17,7 @@
 .eggs
 .vscode
 .tox
+submit_test/
```

Comment:
Ahh I see your point. Yes, I think your solution sounds reasonable. I'm surprised that this hasn't come up before; @sphuber is this the only test which is executing a `dry_run` and comparing the resulting script?

Reply:
There are other tests that run a `dry_run`, and so create this folder, but they don't compare the input script.

Reply:
Ok, I see, but the question is what do you do with all these newly created test folder/files so that they are not shown on git? I actually don't remember these appearing for me after running tests, are they deleted automatically somehow?

Reply:
Hey @sphuber , sorry for the insistence, I just want to make sure there is no already standardized way in which we are dealing with this issue before I propose my own.

To recap the problem: the test needs to run a `dry_run`; this generates a folder with files; this files are left polluting the environment (in this case, the modification aims at removing it from files detected by git). What do other tests that use `dry_run` do? They delete the files manually? Is there a `fixture` that takes care of deleting this folder at the end of the tests?

Reply:
There are two test files that run `CalcJob` dry runs: `tests/engine/test_launch.py` and `tests/engine/test_calc_job.py`. The former cleans up after itself, which is why you never saw the directories before. The `TestLaunchersDryRun` class deletes the `submit_test` folder in its `tearDownClass` method:

https://github.com/aiidateam/aiida-core/blob/a46851972d3bcdb9143801f8199e8c3d17ec645a/tests/engine/test_launch.py#L156-L169

For now, @unkcpz can maybe use this in the `test_calc_job.py` file as well. At some point we can find a cleaner fixture to do this that makes it easy to reuse, but since these test files are still using old-style, maybe we leave that for another time.

Reply:
Thanks @sphuber !

Is it ok then to use some of these at templates to add a cleanup part to your test and remove this from `.gitignore` @unkcpz ?

Reply:
Ah! I see your point! I can do add the fixture here and remove the line from `.gitignore`.